packages/components/date-input/src/use-date-input.ts

```diff
@@ -184,6 +186,48 @@ export function useDateInput<T extends DateValue>(originalProps: UseDateInputPro
     isInvalid: ariaIsInvalid,
   } = useAriaDateField({...originalProps, label, validationBehavior, inputRef}, state, domRef);
 
+  const stringFormatter = useLocalizedStringFormatter(intlMessages) as any;
+
+  if (props.minValue != undefined && validationDetails.rangeUnderflow) {
+    const indexInValidationErrors: number =
+      Number(validationDetails.badInput) +
+      Number(validationDetails.customError) +
+      Number(validationDetails.patternMismatch);
```

Comment:
do we need to cater `rangeUnderflow` as well?

Reply:
I don't think we need `rangeUnderflow` here as only `badInput`, `customError` and `patternMatch` errors appears before `rangeUnderflow` error in validationErrors array. Hence, if some or all of them exist they will shift the index of `rangeUnderflow` in validationErrors. 

But in case of `rangeOverflow` as we have `badInput`, `customError` , `patternMatch`  and `rangeUnderflow` errors before it in validationErrors array, we need to consider all of them.

Reply:
So I assumed that the errors in the `validationField` will be populated according to the occurrence of the property in [ValidityState](https://developer.mozilla.org/en-US/docs/Web/API/ValidityState).
But seems like I made a wrong assumption, as Errors related to `Bad Input` are populated in validationError after `rangeUnderflow` and `rangeOverflow` by RA .([ref](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-stately/datepicker/src/utils.ts#L51-L61)).

Also, while scrapping through RA code, I understood that if ValidationError has `rangeUnderflow`/`rangeOverflow` then it can only have `badInput` error. 
Now, as we might not consider the order of errors in ValidityState as order in ValidityError, it might be bit tricky to find out which error in ValidationError list.

It is clear that if exists in ValidationError, the `rangeOverflow` and `rangeUnderflow` will be its first element. So should we edit the first value in the list?

The reason why the `rangeOverflow` and `rangeUnderflow` will always be first element:
* ValidationErrors can either be obtained from controlledError, serverError, clientError, builtinValidation, currentValidity [(ref)](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-stately/form/src/useFormValidationState.ts#L145-L147) 
* controlledError, serverError, clientError and currentValidity => All of these does not have true for the ValidityState `rangeUnderflow`, `rangeOverflow` and `BadInput`. 
* `rangeOverflow` / `rangeUnderflow` happens to be in the validation errors list if it is obtained from builtinValidation.
* The order in which the errors are populated for builtinValidation is [this](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-stately/datepicker/src/utils.ts#L51-L61).
* Hence, the `rangeOverflow` / `rangeUnderflow` appears to be the first element if they do exist.

Reply:
Updated the following changes in 7838f48cbab983cc2c9b0c91356af5165e72f237
* Added support for badInput message with i18n.
* Modifying the validationError according exactly how they were populated by RA